app/frontend/src/features/map/MapPage.tsx

```diff
@@ -11,27 +12,39 @@ import { MAP_PAGE } from "./constants";
 import addGuidesToMap from "./guides";
 import addPlacesToMap from "./places";
 
-const useStyles = makeStyles((theme) => ({
-  container: {
-    display: "flex",
-    flexDirection: "column",
-    height: `calc(100vh - ${theme.shape.navPaddingMobile})`,
-    paddingBlockEnd: theme.spacing(2),
-    [theme.breakpoints.up("md")]: {
-      height: `calc(100vh - ${theme.shape.navPaddingDesktop})`,
+const useStyles = (windowHeight: number) =>
+  makeStyles((theme) => ({
+    container: {
+      display: "flex",
+      flexDirection: "column",
+      height: `calc(${windowHeight}px - ${theme.shape.navPaddingMobile})`,
+      paddingBlockEnd: theme.spacing(2),
+      [theme.breakpoints.up("md")]: {
+        height: `calc(100vh - ${theme.shape.navPaddingDesktop})`,
+      },
     },
-  },
-  root: {
-    border: "1px solid black",
-  },
-}));
+    root: {
+      border: "1px solid black",
+    },
+  }));
 
 export default function MapPage() {
   const history = useHistory();
 
   const location = useLocation();
 
-  const classes = useStyles();
+  const [windowHeight, setWindowHeight] = useState(window.innerHeight);
+
+  //Update on orientation change
+  useLayoutEffect(() => {
+    const updateWindowHeight = () => {
+      setWindowHeight(window.innerHeight);
+    };
+    window.addEventListener("resize", updateWindowHeight);
+    return () => window.removeEventListener("resize", updateWindowHeight);
+  }, []);
```

Comment:
If we are going down the JS code route... I think it'd be worth extracting this into a custom hook as I can see this being one of those utilities that would be re-usable for other components that need this

Reply:
Also, we should add a debounce around the `updateWindowHeight` if sticking with this, otherwise the `resize` event could fire too often and cause too many unnecessary re-renders